File: backend/logsheet/service/RoleMenuPermissionService.java
package com.sunbeam.logsheet.service;

import java.util.List;


import com.sunbeam.logsheet.DTO.ApiResponse;
import com.sunbeam.logsheet.DTO.RoleMenuPermissionDTO;
import com.sunbeam.logsheet.entity.RoleMenuId;


public interface RoleMenuPermissionService {
	 ApiResponse<?> createPermission(RoleMenuPermissionDTO dto);

	    ApiResponse<?> updatePermission(RoleMenuId id, boolean allowed);

	    ApiResponse<?> deletePermission(RoleMenuId id);

	    RoleMenuPermissionDTO getPermission(RoleMenuId id);

	    List<RoleMenuPermissionDTO> getAllPermissions();
}

File: frontend/logsheet_fe/src/services/roleMenuPermissionService.js


import api from "../utils/api";

const BASE = "/role-menu-permissions";


export const createPermission = (dto) => api.post(BASE, dto);


export const updatePermission = (roleId, menuItemId, allowed) =>
  api.put(`${BASE}/${roleId}/${menuItemId}`, null, { params: { allowed } });


export const deletePermission = (roleId, menuItemId) =>
  api.delete(`${BASE}/${roleId}/${menuItemId}`);


export const getPermission = (roleId, menuItemId) =>
  api.get(`${BASE}/${roleId}/${menuItemId}`);


export const getAllPermissions = () => api.get(BASE);
